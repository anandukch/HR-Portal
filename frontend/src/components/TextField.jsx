/* eslint-disable no-unused-vars */
import React from "react";
import { forwardRef } from "react";

/* eslint-disable react/prop-types */
// eslint-disable-next-line react/display-name
export const TextField = forwardRef(({ name, label, type, className, value, onChange = () => {}, error = "", disabled }, ref) => {
    return (
        <span className={className} data-testid="text-field-test-id">
            <label htmlFor={name} data-testid="label-test-id">
                {label}
            </label>
            <input
                data-testid="input-test-id"
                ref={ref}
                name={name}
                disabled={disabled}
                type={type}
                placeholder={label}
                value={value}
                onChange={onChange}
                style={
                    error != ""
                        ? {
                              border: "1px solid red",
                              color: "red",
                          }
                        : {}
                }
            />
            <span
                style={{
                    color: "red",
                }}
            >
                {error}
            </span>
        </span>
    );
});
