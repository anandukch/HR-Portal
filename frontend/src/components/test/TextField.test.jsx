/* eslint-disable no-unused-vars */
import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { TextField } from "../TextField";

/* eslint-disable no-undef */
describe("Check if Text field works properly", () => {
    test("Check if the text field is rendered", () => {
        const { getByTestId } = render(<TextField />);
        const element = getByTestId("text-field-test-id");
        const label = getByTestId("label-test-id");
        const input = getByTestId("input-test-id");

        expect(element).toBeTruthy();
        expect(label).toBeTruthy();
        expect(input).toBeTruthy();
    });

    test("Check if the text field input value is displayed properly", () => {
        const val = "val";
        const { getByTestId } = render(<TextField value={val} />);
        const inp = getByTestId("input-test-id");
        expect(inp.value).toBe(val);
    });

    test("Check if the text field label is displayed properly", () => {
        const label = "label";
        const { getByTestId } = render(<TextField label={label} />);
        const labelElement = getByTestId("label-test-id");
        expect(labelElement.textContent).toBe(label);
    });

    test("Check if the text field onChange is triggered", () => {
        let paramVal;
        const onChange = jest.fn((e) => {
            paramVal = e.target.value;
        });
        const { getByTestId } = render(<TextField onChange={onChange} />);
        const input = getByTestId("input-test-id");
        const onChangeVal = "changed";
        fireEvent.change(input, { target: { value: onChangeVal } });
        expect(onChange).toHaveBeenCalled();
        expect(paramVal).toBeTruthy();
        expect(paramVal).toBe(onChangeVal);
    });

    test("Check if snapshots are mapped properly", () => {
        const onChange = jest.fn();
        const label = "label";
        const val = "val";
        const { asFragment } = render(<TextField onChange={onChange} label={label} value={val} type="text" name="name" />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("check if error message has been rendered", () => {
        const { getByText } = render(<TextField error="Error" />);
        const element = getByText("Error");
        expect(element.textContent).toBe("Error");
    });

    test("check if input is disabled on passing diabled flag", () => {
        const { getByTestId } = render(<TextField disabled />);
        const input = getByTestId("input-test-id");
        expect(input).toBeDisabled();
    });

    test("check if input type has bee assigned properly", () => {
        const type = "password";
        const { getByTestId } = render(<TextField type={type} />);
        const input = getByTestId("input-test-id");
        expect(input.type).toBeTruthy();
        expect(input.type).toBe(type);
    });
});
