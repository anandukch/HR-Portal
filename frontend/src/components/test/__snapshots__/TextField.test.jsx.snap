// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Text field works properly Check if snapshots are mapped properly 1`] = `
<DocumentFragment>
  <span
    data-testid="text-field-test-id"
  >
    <label
      data-testid="label-test-id"
      for="name"
    >
      label
    </label>
    <input
      data-testid="input-test-id"
      name="name"
      placeholder="label"
      type="text"
      value="val"
    />
    <span
      style="color: red;"
    />
  </span>
</DocumentFragment>
`;
